cmake/DyninstBoost.cmake

```
@@ -0,0 +1,380 @@
+
+set(_boost_defines)
+
+# Disable auto-linking
+list(APPEND _boost_defines BOOST_ALL_NO_LIB=1)
+
+# Disable generating serialization code in boost::multi_index
+list(APPEND _boost_defines BOOST_MULTI_INDEX_DISABLE_SERIALIZATION)
+
+# There are broken versions of MSVC that won't handle variadic templates correctly
+# (despite the C++11 test case passing).
+if(MSVC)
+    list(APPEND _boost_defines BOOST_NO_CXX11_VARIADIC_TEMPLATES)
+endif()
+
+set(Boost_DEFINES
+    ${_boost_defines}
+    CACHE STRING "Boost compiler defines")
+add_compile_definitions(${Boost_DEFINES})
+
+# -------------- INTERNALS ----------------------------------------------------
+
+# Disable Boost's own CMake as it's known to be buggy NB: This should not be a cache
+# variable
+set(Boost_NO_BOOST_CMAKE ON)
+
+# The required Boost library components NB: These are just the ones that require
+# compilation/linking This should _not_ be a cache variable
+set(_boost_components atomic chrono date_time filesystem system thread timer)
+
+if(NOT BUILD_BOOST)
+    find_package(Boost ${Boost_MIN_VERSION} QUIET COMPONENTS ${_boost_components})
+endif()
+
+# -------------- SOURCE BUILD -------------------------------------------------
+
+if(Boost_FOUND AND NOT BUILD_BOOST)
+    # Force the cache entries to be updated Normally, these would not be exported.
+    # However, we need them in the Testsuite
+    set(Boost_INCLUDE_DIRS
+        ${Boost_INCLUDE_DIRS}
+        CACHE PATH "Boost include directory" FORCE)
+    set(Boost_LIBRARY_DIRS
+        ${Boost_LIBRARY_DIRS}
+        CACHE PATH "Boost library directory" FORCE)
+    set(Boost_INCLUDE_DIR
+        ${Boost_INCLUDE_DIR}
+        CACHE PATH "Boost include directory" FORCE)
+elseif(NOT Boost_FOUND AND STERILE_BUILD)
+    rocprofiler_systems_message(
+        FATAL_ERROR "Boost not found and cannot be downloaded because build is sterile.")
+elseif(NOT BUILD_BOOST)
+    rocprofiler_systems_message(
+        FATAL_ERROR
+        "Boost was not found. Either configure cmake to find Boost properly or set BUILD_BOOST=ON to download and build"
+        )
+else()
+    rocprofiler_systems_add_option(BOOST_LINK_STATIC "Link to boost libraries statically"
+                                   ON)
+    # If we didn't find a suitable version on the system, then download one from the web
+    rocprofiler_systems_add_cache_option(
+        BOOST_DOWNLOAD_VERSION "Version of boost to download and install" STRING "1.79.0")
+
+    # If the user specifies a version other than BOOST_DOWNLOAD_VERSION, use that version.
+    if(${BOOST_DOWNLOAD_VERSION} VERSION_LESS ${Boost_MIN_VERSION})
+        rocprofiler_systems_message(
+            FATAL_ERROR
+            "Boost download version is set to ${BOOST_DOWNLOAD_VERSION} but Boost minimum version is set to ${Boost_MIN_VERSION}"
+            )
+    endif()
+
+    rocprofiler_systems_message(
+        STATUS "Attempting to build ${BOOST_DOWNLOAD_VERSION} as external project")
```

```suggestion
        STATUS "Attempting to build BOOST(${ROCPROFSYS_BOOST_DOWNLOAD_VERSION}) as external project")
```
Updated log to match style of other components